blockchain/system/rebalance.go

```diff
@@ -68,95 +130,126 @@ func (caller *Kip103ContractCaller) CallContract(ctx context.Context, call klayt
 	}
 	return nil
 }
 
-func (result *kip103result) fillNewbie(contract *rebalance.TreasuryRebalanceCaller) error {
-	numNewbieBigInt, err := contract.GetNewbieCount(nil)
+func (result *rebalanceResult) fillAllocated(contract RebalanceCaller) error {
+	numNewbieBigInt, err := contract.GetAllocatedCount(nil)
 	if err != nil {
-		logger.Error("Failed to get NewbieCount from TreasuryRebalance contract", "err", err)
+		logger.Error("Failed to get AllocatedCount from TreasuryRebalance contract", "err", err)
 		return nil
 	}
 
 	for i := 0; i < int(numNewbieBigInt.Int64()); i++ {
-		ret, err := contract.Newbies(nil, big.NewInt(int64(i)))
+		ret, err := contract.Allocateds(nil, big.NewInt(int64(i)))
 		if err != nil {
-			logger.Error("Failed to get Newbies from TreasuryRebalance contract", "err", err)
+			logger.Error("Failed to get Allocateds from TreasuryRebalance contract", "err", err)
 			return err
 		}
-		result.Newbie[ret.Newbie] = ret.Amount
+
+		result.Allocated[ret.Addr] = ret.Amount
 	}
 	return nil
 }
 
-func (result *kip103result) totalRetiredBalance() *big.Int {
+func (result *rebalanceResult) totalZeroedBalance() *big.Int {
 	total := big.NewInt(0)
-	for _, bal := range result.Retired {
+	for _, bal := range result.Zeroed {
 		total.Add(total, bal)
 	}
 	return total
 }
 
-func (result *kip103result) totalNewbieBalance() *big.Int {
+func (result *rebalanceResult) totalAllocatedBalance() *big.Int {
 	total := big.NewInt(0)
-	for _, bal := range result.Newbie {
+	for _, bal := range result.Allocated {
 		total.Add(total, bal)
 	}
 	return total
 }
 
-// RebalanceTreasury reads data from a contract, validates stored values, and executes treasury rebalancing (KIP-103).
+// RebalanceTreasury reads data from a contract, validates stored values, and executes treasury rebalancing (KIP-103, KIP-160).
 // It can change the global state by removing old treasury balances and allocating new treasury balances.
 // The new allocation can be larger than the removed amount, and the difference between two amounts will be burnt.
-func RebalanceTreasury(state *state.StateDB, chain backends.BlockChainForCaller, header *types.Header) (*kip103result, error) {
+func RebalanceTreasury(state *state.StateDB, chain backends.BlockChainForCaller, header *types.Header) (*rebalanceResult, error) {
 	var (
-		result = newKip103Receipt()
-		c      = NewKip103ContractCaller(state, chain, header)
+		err    error
+		caller RebalanceCaller
+
+		result   = newRebalanceReceipt()
+		isKIP160 = chain.Config().IsKIP160ForkBlock(header.Number)
+		isKIP103 = chain.Config().IsKIP103ForkBlock(header.Number)
 	)
 
-	caller, err := rebalance.NewTreasuryRebalanceCaller(chain.Config().Kip103ContractAddress, c)
+	if isKIP160 {
+		caller, err = rebalance.NewTreasuryRebalanceV2Caller(chain.Config().Kip160ContractAddress, backends.NewBlockchainContractBackend(chain, nil, nil))
+	} else if isKIP103 {
+		caller, err = NewKip103ContractCaller(state, chain, header)
+	}
```

Comment:
```suggestion
	} else if isKIP103 {
		caller, err = NewKip103ContractCaller(state, chain, header)
	} else {
	        return nil, ErrRebalanceIncorrectBlock
	}
```

Reply:
`ErrRebalanceIncorrectBlock` indicates the incorrect behavior of the contract, so instead, it returned the custom error message. Reflected at https://github.com/klaytn/klaytn/pull/2122/commits/8a7a3b3ba70bf188d39547449ad20e97ec644735.